osquery/main/windows/daemon.cpp

```diff
@@ -70,6 +75,90 @@ void DebugPrintf(const char* fmt, ...) {
   va_end(vl);
 }
 
+/*
+ * Parses arguments for the Windows service. Arguments to the Windows service
+ * can be passed in two ways: via sc.exe (manual start) or via binPath
+ * (automated start). Unfortunately, both use different methods of getting the
+ * command line arguments. Manual start uses the argc and argv provided by
+ * ServiceMain whereas automated start requires manual parsing of
+ * GetCommandLine()
+ */
+class ServiceArgumentParser {
+ public:
+  ServiceArgumentParser(DWORD argc, LPSTR* argv) {
+    if (argc > 1) {
+      for (DWORD i = 0; i < argc; i++) {
+        args_.push_back(argv[i]);
```

Comment:
Are you appending pointers you don't control? Then `delete`ing them in the `cleanArgs` method?